### PR TITLE
[doc] Add "Alternatives" for Bazel and ROS 2 workflows; mention ApexAI/rules_ros and mvukov/rules_ros2

### PR DESCRIPTION
**Pull request by EricCousineau-TRI**
Per internal discussion.

Filed issue for RFC and possible symmetry:
- https://github.com/ApexAI/rules_ros/issues/5
- https://github.com/mvukov/rules_ros2/pull/24

- - -
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/194)

### CODE REVIEW

**Review by EricCousineau-TRI — COMMENTED** [review]
+@sloretz for review, please!
\cc @cottsay 

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/194#-:-NHAd61sE6XVXV8v_AYS:bpp5bcs)* status: 0 of 3 files reviewed, all discussions resolved (waiting on @sloretz)

**Comment by mvukov** [comment]
Here is a yet another alternative https://github.com/mvukov/rules_ros2/ ;).

**Review by EricCousineau-TRI — COMMENTED** [review]
Done, thanks! I've tried to add in a summary here. If you have a chance, could you take a look to see if it's fair / accurate?

And it cool if I file issue against your repo for similar? If you wanna link here, or add your own analysis, I'm good either way!

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/194#-:-NHLmzBcBqQmJ0DTupxy:b-1vtwpa)* status: 0 of 3 files reviewed, all discussions resolved (waiting on @sloretz)

File: bazel_ros2_rules/ros2/README.md [hunk 1]
```diff
@@ -114,7 +114,83 @@ provided (such as Bazel's `$TEST_TMPDIR`).
...
+[`rules_docker`](https://github.com/bazelbuild/rules_docker)).
+
+The benefit from this approach is being able to fully control the source build,
+configuration flags, etc., for ensuring a build is consistent and easily
+reconfigurable.
+
+The (present) possible con to this approach is scalability. You must presently
+provide your own set of external `{repo}.BUILD.bazel` rules that reflect the
+ament operations expressed in CMake. Until the `ament` / ROS 2 build ecosystem
+provides scaffolding or affordances for Bazel tooling, and until pacakges
```
Comment (mvukov) [inline bazel_ros2_rules/ros2/README.md:144]: Nit: packages.

**Review by mvukov — COMMENTED** [review]
Very nice summary, thanks!

**Comment by mvukov** [comment]
> And it cool if I file issue against your repo for similar? If you wanna link here, or add your own analysis, I'm good either way!

Sure, you can either open an issue or make a PR.

**Review by EricCousineau-TRI — COMMENTED** [review]
Aye, I'll make a quick PR.

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/194#-:-NHLpL9c9cWTgL-aWcoZ:bcq6zri)* status: 0 of 3 files reviewed, all discussions resolved (waiting on @sloretz)

___
*[`bazel_ros2_rules/ros2/README.md` line 144 at r2](https://reviewable.io/reviews/robotlocomotion/drake-ros/194#-NHAbieMxeaUI8SqGWPI-r2-144:-NHLpJ2g0e7pRpMbaRyY:bxsser9) ([raw file](https://github.com/robotlocomotion/drake-ros/blob/2ca6ff8883bb837ad958c0d0aca1b5115b01de73/bazel_ros2_rules/ros2/README.md#L144)):*
<details><summary><i>Previously, mvukov (Milan Vukov) wrote…</i></summary><blockquote>

Nit: packages.
</blockquote></details>

Done 0ef58e9, thanks!

**Comment by EricCousineau-TRI** [comment]
--> https://github.com/mvukov/rules_ros2/pull/24

**Comment by kilian-funk** [comment]
@EricCousineau-TRI This looks good to me. Thanks for referencing the ApexAI/rules_ros here.

**Comment by EricCousineau-TRI** [comment]
See also discussions in https://discourse.ros.org/t/adding-bazel-as-an-official-alternative-build-system-for-ros2-iron/28476

**Review by sloretz — CHANGES_REQUESTED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/194)* status: all files reviewed, 1 unresolved discussion (waiting on @EricCousineau-TRI)

___
*[`bazel_ros2_rules/ros2/README.md` line 186 at r3](https://reviewable.io/reviews/robotlocomotion/drake-ros/194#-NI-qgik8P3iWuQc-hVf:-NI-qgik8P3iWuQc-hVg:buz6us6) ([raw file](https://github.com/robotlocomotion/drake-ros/blob/0ef58e9b1187388655f315b5c48444ab45dbc478/bazel_ros2_rules/ros2/README.md#L186)):*
> ```Markdown
> - Writing Python bindings in `pybind11` against other C++ libraries that have
>   their own bindings - e.g., [Drake](https://drake.mit.edu/).
>   - There is *some* affordance for leveraging `pybind11` against ROS 2 RMW
> ```

I don't see anything pybind11 specific in `bazel_ros2_rules`. Is this actually a feature of this approach? I know we use pybind11 in `drake_ros_*` but that seems unrelated to the generated bazel rules.

```
grep bazel_ros2_rules/ -r -e "pybind"
```

**Review by EricCousineau-TRI — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/194)* status: 2 of 3 files reviewed, 1 unresolved discussion (waiting on @sloretz)

___
*[`bazel_ros2_rules/ros2/README.md` line 186 at r3](https://reviewable.io/reviews/robotlocomotion/drake-ros/194#-NI-qgik8P3iWuQc-hVf:-NI2cxJhCUYJSgZX7kQj:b-qo0wcf) ([raw file](https://github.com/robotlocomotion/drake-ros/blob/0ef58e9b1187388655f315b5c48444ab45dbc478/bazel_ros2_rules/ros2/README.md#L186)):*
<details><summary><i>Previously, sloretz (Shane Loretz) wrote…</i></summary><blockquote>

I don't see anything pybind11 specific in `bazel_ros2_rules`. Is this actually a feature of this approach? I know we use pybind11 in `drake_ros_*` but that seems unrelated to the generated bazel rules.

```
grep bazel_ros2_rules/ -r -e "pybind"
```
</blockquote></details>

Done. You're right! But I still want to emphasize the examples portion, so I've rescoped this statement as such.

**Review by sloretz — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/194)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @EricCousineau-TRI)

**Comment by EricCousineau-TRI** [comment]
Thanks!

**Pull request merged**